tasks_week3_Shai_Axelrod/Zoo/include/Animal.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+#include "Location.h"
+#include "Direction.h"
+/**
+ * @class Animal
+ * @brief Represents an animal in the simulation.
+ */
+class Animal {
+protected:
+    int steps; /**< The number of steps the takes in each move. */
+    std::string name; /**< The name of the animal. */
+    Location location; /**< The current location of the animal. */
+    bool inMoveMode; /**< Whether the animal is currently in move mode. */
+    Direction direction; /**< The direction the animal is facing. */
+
+    /**
+     * @brief Returns the type of the animal.
+     * @return The type of the animal as a string.
+     */
+    virtual std::string getType() const = 0;
+
+    /**
+     * @brief Randomly selects a new direction for the animal to move.
+     */
+    virtual void ballotNewDir() = 0;
+
+public:
+    /**
+     * @brief Constructor for the Animal class.
+     * @param name The name of the animal.
+     * @param location The initial location of the animal.
+     */
+    Animal(const std::string& name, const Location& location);
+
```

Comment:
no use of virtual destructor 
check this
https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors

Reply:
again, no need to use virtual here. The inheriting classes have no cleanup to do.